src/gui/MainWindow.cpp

```diff
@@ -276,6 +276,12 @@ MainWindow::MainWindow(Settings settings, QWidget *parent)
         settings_to_ui(settings_, *ui_, module_display_);
     });
 
+    connect(ui_->patterns, &QComboBox::activated, this, [this]() {
+        save_settings();
```

Comment:
Hmm now that I am thinking about it, it shouldn’t be necessary - but there should be a forced call to `ui_to_settings` on all the module widgets I think. Reasoning is to avoid losing changes to settings on switching patterns.

Reply:
Alright. In this case, I don't mind saving